omegaconf/_utils.py

```diff
@@ -553,17 +553,21 @@ def none_as_any(t: Optional[Type[Any]]) -> Union[Type[Any], Any]:
     elif isinstance(obj, Container):
         if isinstance(obj, Node):
             ref_type = obj._metadata.ref_type
-        if not is_structured_config(ref_type):
-            kt = none_as_any(obj._metadata.key_type)
-            vt = none_as_any(obj._metadata.element_type)
+        is_optional = obj._is_optional()
+        kt = none_as_any(obj._metadata.key_type)
+        vt = none_as_any(obj._metadata.element_type)
+        if ref_type is Any and kt is Any and vt is Any and obj._value() != "???":
+            pass
```

Comment:
1. This is easier to read:
```suggestion
            ref_type = Any
```

2. Missing Container has their own ref type.

```python
ref_type(DictConfig(content="???", element_type=str, key_type=int)) == Optional[Dict[str, int]]
```
Make sure there is a test for this (and for missing ListConfig).

3. Also containers that are none. Use OmegaConf.is_missing() and OmegaConf.is_none() to test that instead of comparing to '???' and None directly.

Reply:
> Also containers that are none. Use OmegaConf.is_missing() and OmegaConf.is_none() to test that instead of comparing to '???' and None directly.
Well should it  be obj._is_missing() instead because obj is already the node to check?

Reply:
You can use either.